containers/Signup/Signup.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from "react";
+import { ScrollView, StatusBar, Text, TextInput, TouchableOpacity, View } from "react-native";
+import { styles } from "./style";
+
+export const Signup = ({ navigation }: { navigation: any }) => {
+
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [isError, setIsError] = useState(false);
+    const [emailErrorMessage, setEmailErrorMessage] = useState("");
+    const [isPassError, setIsPassError] = useState(false);
+    const [passErrorMessage, setPassErrorMessage] = useState("");
+    
+    const handleSignupPress = () => {
+        navigation.navigate({
+            name: "Dashboard",
+            params: {
+                userName: email
+            }
+        });
+    }
+    const handleLoginPress = () => {
+        navigation.navigate("Login");
+    }
+
+    const validateSymbols = (email: string) => {
+        let re = /[ `!#$%^&*()_+\-=\[\]{};':"\\|,<>\/?~]/;
+          return re.test(email);
+    };
+    const validateAtss = (email: string) => {
+        let count = 0;
+        for(let i=0; i<email.length; i++){
+            if(email.charAt(i).includes("@"))
+                count++;
+        }
+        return count;
+    };
+    const validatePass = (password: string) => {
+        let message = "";
+        if(password.length < 8) message = "* Minimum length is 8 \n"
+        let check1 = /[0-9a-zA-z+!\/]/;
+        if(!check1.test(password)) message += "* Only letters, numbers and symbols are required \n"
+        else {
+            let check2 = /[0-9]/
+            if(!check2.test(password)) message += "* Atleast 1 letter is required \n"
+
+            let check3 = /[+!\/]/
+            if(!check3.test(password)) message += "* Atleast 1 symbol is required \n"
+
+        }
+        return message;
+    }
+
+    return (
+        <View style={styles.signContainer}>
+            <StatusBar barStyle="light-content" />
+
+            <Text style={styles.bannerCreate}>Create Account</Text>
+
+            <TextInput
+            value={email}
+            placeholder="Email"
+            placeholderTextColor="#aaa"
+            keyboardType='web-search'
+            autoCapitalize="none"
+            autoFocus={false}
+            onChangeText={val => {
+
+                if(validateAtss(val)>1){
```

Comment:
Interesting approach to validate inside `onChangeText` directly. Wouldn't there be rendering repercussions for running logic on key stroke?

Reply:
Yes there would rendering repercussions of logic on every key stroke. 

I wanted to keep the validation logic seperate as Validator.tsx and then callback from there to add validations and display corresponding message.